### DISCUSSION TITLE
LAG support

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#658] [Features]
Can we add Link Aggregation Group (LAG) support to Netlab?

I've tried creating a plugin, but I think this requires the creation of a new type of interface and is better handled natively

**Comment by ipspace** [comment]
> Can we add Link Aggregation Group (LAG) support to Netlab?

I knew that someone would inevitably open this Pandora's box ;)) And let me guess, in a few weeks we'll be at the "but we also need MLAG" stage, right?

IIRC, netlab enables LACP forwarding on Linux bridges after starting the lab, but it might make sense to limit this to P2P links with _libvirt_ or _clab_ provider (containerlab can handle LAG, right?)
 
> I've tried creating a plugin, but I think this requires the creation of a new type of interface and is better handled natively

Agree. It might be best if you'd start with a description of the data model (also considering potential desire to add MLAG); fixing the code after finding out that the data model is wrong tends to be a bit time-consuming.

FWIW, apart from reviewing the code and creating configuration templates after the module is done, I'm not touching this one after my wonderful experience with the VLAN module ;))

**Comment by ssasso** [comment]
I addressed both LAG and MCLAG using custom templates :-)

* https://github.com/ssasso/netsim-topologies/tree/main/dell_os10_vlt
* https://github.com/ssasso/netsim-topologies/tree/main/arista_mlag

But for sure having it fully integrated would work better.

I can help with the data model review, and with the testing/platform support, when needed - if you decide to open this pandora's box ;)

**Comment by jbemmel** [comment]
Implemented https://github.com/ipspace/netlab/pull/1350
